### PR TITLE
Ge 120

### PR DESCRIPTION
**Pull request by gh-23378**

### CODE REVIEW

File: src/store/timetravel.js [hunk 1]
```diff
@@ -65,6 +61,15 @@ export default {
...
       if (window.api) { window.api.config.onChange(); }
+      const newState = args[0];
+
+      // non timetravel props
+      // TODO: ask brian about config syntax for this
+      newState.application.scale = this.state.application.scale;
+      newState.project.view = this.state.project.view;
+      newState.project.grid = this.state.project.grid;
+      newState.project.previous_story = this.state.project.previous_story;
+
```
Comment (bgschiller) [inline src/store/timetravel.js:60]: what is `newState` being used for here?

Also, were you looking for something like `newState = omit(this.state, 'timetravel')`?

File: src/store/timetravel.js [hunk 2]
```diff
@@ -1,11 +1,7 @@
 const serializeState = (state) => {
-  const scaleX = state.application.scale.x;
-  const scaleY = state.application.scale.y;
-
   const clone = JSON.parse(JSON.stringify(state));
-  clone.application.scale.x = scaleX;
-  clone.application.scale.y = scaleY;
 
+  // TODO: update these to be IDs and not references, then update importcode and serealization code
```
Comment (bgschiller) [inline src/store/timetravel.js]: is this still todo?

File: src/store/utilities/importModel.js [hunk 3]
```diff
@@ -1,95 +1,96 @@
...
-    payload.data.stories.forEach((story) => {
-        story.geometry_id = story.geometry.id;
-        delete story.geometry;
+    geometry[i].vertices = geometry[i].vertices.map((v) => {
+      return {
+        id: v.id,
+        x: v.x,
+        y: v.y,
+        X: v.x,
+        Y: v.y
```
Comment (bgschiller) [inline src/store/utilities/importModel.js]: Why are we duplicating these to be both lower and upper case?

**Review by bgschiller — COMMENTED** [review]
It looks like the new code doesn't work for importing old floorplans. Is that something we need to handle? Looks good otherwise 👍

File: src/store/utilities/importModel.js (hunk 3 above)
Comment (gh-23378) [reply src/store/utilities/importModel.js]: This used to be required for clipper, but can actually be removed now. Good catch!

File: src/store/timetravel.js (hunk 1 above)
Comment (gh-23378) [reply src/store/timetravel.js:60]: This ignores properties on the state which we do not want affected by timetravel

**Pull request merged**